x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/editor_frame.test.tsx

```diff
@@ -294,7 +295,9 @@ describe('editor_frame', () => {
 
       await waitForPromises();
 
-      mockVisualization.initialize.mock.calls[0][0].addNewLayer();
```

Comment:
If I'm writing the test like this:
```tsx
    it.only('should add new layer on active datasource on frame api call', async () => {
      const initialState = { datasource2: '' };
      mockDatasource2.initialize.mockReturnValue(Promise.resolve(initialState));
      await act(async () => {
        mount(
          <EditorFrame
            {...getDefaultProps()}
            visualizationMap={{
              testVis: mockVisualization,
            }}
            datasourceMap={{
              testDatasource: mockDatasource,
              testDatasource2: mockDatasource2,
            }}
            initialDatasourceId="testDatasource2"
            initialVisualizationId="testVis"
            ExpressionRenderer={expressionRendererMock}
          />
        );
      });

      act(() => {
        mockVisualization.initialize.mock.calls[0][0].addNewLayer();
      });

      expect(mockDatasource2.insertLayer).toHaveBeenCalledWith(initialState, expect.anything());
    });

```

Then it passes for me and I don't get warnings anymore. It seems like an async act around the mounting is enough for react to pick up the async changes and we don't need a separate call to a helper function like `waitForPromises`